vkrunner/vr-script.c

```diff
@@ -2027,6 +2027,25 @@ vr_script_load(const struct vr_config *config,
         return script;
 }
 
+struct vr_script *
+vr_script_load_from_string(const struct vr_config *config,
+                           const char *string)
+{
+        struct vr_script *script;
+        FILE *f = fmemopen((void *)string, strlen(string), "r");
```

Comment:
It looks like this function isn’t available on Windows. I did a quick Google and as far as I can tell it’s not available on the Android NDK either? Maybe for the sake of portability it would be better to avoid it.

Perhaps we could split `load_script_from_stream()` into `begin_load()`, `load_from_stream()`,  and `end_load()` and then for the string version we could replace `load_from_stream()` with a function that looks for lines in the string and copies them into `data.line` before calling `process_line()`. Or if you have a preference for a different way to do it, of course feel free to try it.

Reply:
How about just removing handling the file stream from `vr-script.c` file and just read all file contents before starting the script parse?

For example,
1. read all file contents in `process_script()` of `vr-execute.c`.
2. `vr_script_load()` just receives the string and parses it.

Reply:
If we do that we still have write the code to split the string into lines and parse the lines one at a time. It doesn’t seem like it should be much more effort to keep the stream handling code as well. Unless there is a good reason to remove it I’d prefer if we could keep it.

Reply:
Done.